### PR TITLE
Reset failureCount to zero

### PR DESCRIPTION
**Pull request by adibfirman**
Hello,
I found some problem when i am using `failureCount` it's not reset to zero as your readme say.

Thanks

### CODE REVIEW

File: src/queryCache.js [hunk 1]
```diff
@@ -316,6 +317,7 @@ export function makeQueryCache() {
               try {
                 const data = await tryFetchData(queryFn, ...args)
                 if (query.cancelled) return reject(query.cancelled)
+                dispatch({ type: actionResetFailureCount })
```
Comment (tannerlinsley) [inline src/queryCache.js]: This is unnecessary with the suggested changes below.

File: src/queryCache.js [hunk 2]
```diff
@@ -21,6 +21,7 @@ const actionFetch = {}
 const actionSuccess = {}
 const actionError = {}
 const actionSetData = {}
+const actionResetFailureCount = {}
```
Comment (tannerlinsley) [inline src/queryCache.js]: This is unnecessary with the suggested changes below.

File: src/queryCache.js [hunk 3]
```diff
@@ -419,6 +421,11 @@ export function makeQueryCache() {
 
 export function defaultQueryReducer(state, action) {
   switch (action.type) {
+    case actionResetFailureCount:
+      return {
+        ...state,
+        failureCount: 0
+      }
```
Comment (tannerlinsley) [inline src/queryCache.js]: Instead of using this new action, just updated the success action below.

Comment (tannerlinsley) [reply src/queryCache.js]: The `actionSuccess` handler can look like this: 

```diff
case actionSuccess:
      return {
        ...state,
        status: statusSuccess,
        data: action.data,
        error: null,
        isStale: false,
        isFetching: false,
        canFetchMore: action.canFetchMore,
        updatedAt: Date.now(),
+        failureCount: 0
      }
```

Comment (adibfirman) [reply src/queryCache.js]: that s is a neat, thanks for reviewing my code :pray:

**Comment by deevus** [comment]
Love an easy fix :)

**Pull request merged**